### DISCUSSION TITLE
Add config to prevent log rejections in the console

### DISCUSSION THREAD

**Discussion by bruno-vinicius** [TanStack/query#1292] [General] [answered]
Hello,

When I was testing react-query, I was a little confused to find out what was firing a log (from my Promise.reject) in the console, despite the one that fetch already do. Then I realised that the react-query [logs rejections by default](https://github.com/tannerlinsley/react-query/blob/master/src/core/query.ts#L412). Would it be interesting to put a configuration to prevent it being logged in the browser?

I say this thinking about applications that have error tracking and that may or may not not want everything to be logged into the console by default (since fetch already logs - but don't throws - request errors).

Print from console:
![image](https://user-images.githubusercontent.com/4923516/99301089-33f07200-282c-11eb-9e7c-3114680289e8.png)

My code as example:

```
const defaultQueryFn = () => {
  return fetch(
    `http://localhost:8011/api/my-api`
  ).then(async response => {
    const data = await response.json()
    if (response.ok) {
      return data
    } else {
      return Promise.reject(response)
    }
  })
};
```

**Comment by ctwillie** [comment, marked as answer]
@bruno-vinicius You can set a custom logger for  general logs, warnings and errors: See docs on [custom logger](https://react-query.tanstack.com/reference/setLogger).

**Reply by bruno-vinicius** [reply]
Thanks! This was exactly what I needed.
